hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/conf/TestZKConfigurationStore.java

```diff
@@ -401,6 +407,34 @@ public void testFailoverAfterRemoveQueue() throws Exception {
+    Assert.assertNull(confStore.retrieve());
+
+    if (!System.getProperty("os.name").startsWith("Windows")) {
+      for (int i = 0; i < 20; ++i) {
+        if (flagFile.exists()) {
+          continue;
+        }
+        Thread.sleep(100);
+      }
+
+      Assert.assertFalse(flagFile.exists());
```

Comment:
can you add the error text to raise in the assert, including the filename.

Reply:
Thanks for the review. I added an error message.


I see some test failures with errors `Metrics source DelegationTokenSecretManagerMetrics already exists!`, that was introduced with this commit (git bisect):

```
d60262fe0092a9b45ee17830b98750fb00b856b1 is the first bad commit
commit d60262fe0092a9b45ee17830b98750fb00b856b1
Author: hchaverri <55413673+hchaverri@users.noreply.github.com>
Date:   Tue Apr 26 09:20:11 2022 -0700

    HADOOP-18167. Add metrics to track delegation token secret manager op… (#4092)

    * HADOOP-18167. Add metrics to track delegation token secret manager operations

 .../AbstractDelegationTokenSecretManager.java      | 121 +++++++++++++++++-
 .../token/delegation/TestDelegationToken.java      | 136 +++++++++++++++++++++
 2 files changed, 252 insertions(+), 5 deletions(-)
```

Reply:
This is fixed with HADOOP-18222.
Please rebase your branch so that UT failures can go away :)